pinot-core/src/main/java/org/apache/pinot/core/query/reduce/StreamingReduceService.java

```diff
@@ -106,15 +104,18 @@ public BrokerResponseNative reduceOnStreamResponse(BrokerRequest brokerRequest,
     return brokerResponseNative;
   }
 
-  private static void processIterativeServerResponse(StreamingReducer reducer, ExecutorService executorService,
```

Comment:
This needs to be reverted so the work is done within `_reduceExecutorService`

Reply:
Thanks. Didn't see that `supplyAsync( lambda, executor)` method. Fixed now